### PR TITLE
[INI] Add extra configuration file for product support

### PR DESCRIPTION
**Pull request by again4you**
The product team wants to update the nnstreamer configuration (i.e.
/etc/nnstreamer.ini). However, the partition is Read-Only so it is not
updatable. Because of this reason, they want another nnstreamer
configuration file to meet their request such as product-specific
restricted elements. This patch newly adds the nnstreamer extension ini
file for product support.

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### New element in nnstreamer.ini  ###
```ini
[common]
extra_config_path=/tmp/nns-api-X6XDF1/nns-product.ini	 # Location of product-specific configuration file.
```

### Related issue
* https://github.com/nnstreamer/nnstreamer/issues/3629

### CODE REVIEW

File: nnstreamer.ini.in [hunk 1]
```diff
@@ -30,4 +31,4 @@ subplugin_priority=@TFLITE_SUBPLUGIN_PRIORITY@
 [filter-aliases]
 trix-engine = @TRIX_ENGINE_ALIAS@
 
-@ELEMENT_RESTRICTION_CONFIG@
+@ELEMENT_RESTRICTION_CONFIG@
```
Comment (jaeyun-jung) [inline nnstreamer.ini.in:34]: plz check end-of-str char

File: tests/common/unittest_common.cc [hunk 2]
```diff
@@ -1664,6 +1664,43 @@ TEST (confCustom, envStr01)
...
+  gchar *dir = g_mkdtemp (fullpath);
+  gchar *filename = g_build_path ("/", dir, "nnstreamer.ini", NULL);
+  const gchar *extra_conf = "/opt/usr/vd/product.ini";
+  const gchar *base_confenv;
+  gchar *confenv;
+  FILE *fp = g_fopen (filename, "w");
+  ASSERT_TRUE (fp != NULL);
+
+  base_confenv = g_getenv ("NNSTREAMER_CONF");
+  confenv = (base_confenv != NULL) ? g_strdup (base_confenv) : NULL;
```
Comment (jaeyun-jung) [inline tests/common/unittest_common.cc:1682]: IMO base_confenv is unnecessary,
`confenv = g_strdup (g_getenv ("NNSTREAMER_CONF"));`

File: tests/common/unittest_common.cc [hunk 3]
```diff
@@ -1664,6 +1664,39 @@ TEST (confCustom, envStr01)
...
+  g_fprintf (fp, "extra_config_path=%s\n", extra_conf);
+  fclose (fp);
+
+  EXPECT_TRUE (g_setenv ("NNSTREAMER_CONF", filename, TRUE));
+  EXPECT_TRUE (nnsconf_loadconf (TRUE));
+
+  EXPECT_TRUE (check_custom_conf ("common", "extra_config_path", extra_conf));
+  g_free (fullpath);
+  g_free (filename);
+  g_remove (filename);
```
Comment (jaeyun-jung) [inline tests/common/unittest_common.cc:1690]: uh..need fix here (use char ptr after free)

Comment (again4you) [reply tests/common/unittest_common.cc:1690]: Oh! I will fix it!

**Review by jaeyun-jung — APPROVED** [review]

**Review by myungjoo — APPROVED** [review]

**Pull request merged**